[CC 1.0.0] Read APIs from File System [4.0.0] (#5145)
wso2/docs-apim: en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-apis-as-immutable-gateway.md

# Deploy APIs as Immutable Gateway

When deploying APIs as an immutable Gateway, the APIs are deployed during startup. After startup, no APIs can be added using apictl. 

> For that, the apictl projects needs to be mounted to the `docker-compose/resources/adapter/artifacts/apis` directory 

Reviewer: ```suggestion
To do this, the apictl projects need to be mounted to the `docker-compose/resources/adapter/artifacts/apis` directory 
```

> (default location) of the adapter. 

Reviewer: ```suggestion
(default location) of the Adapter. 
```
Let's add all component names in Titlecase.

``` bash
apictl init petstore --oas https://petstore.swagger.io/v2/swagger.json
```

!!! note

    Instead of initializing the API project, you can [export the APIs from API Manager]({{base_path}}/install-and-setup/setup/api-controller/managing-apis-api-products/migrating-apis-to-different-environments) as well.

## Step 3 - Copy the project to the resources/enforcer/artifacts/apis directory

The created apictl project needs to be copied to `docker-compose/resources/adapter/artifacts/apis` directory. These
apictl projects could be copied either in directory format or in zipped format. 
The apictl project can be the API artifact exported from the API Manager as well. 
The docker-compose file mounts that directory to the `/home/wso2/artifacts/apis`. 
All the apictl projects in the directory will be deployed during the startup. 

## Step 4 - update config.toml

```
[adapter.server]
    enabled = false
```

!!! note

    If this configuration is set to true, the mounted API projects are deployed during startup. However, they will not have
    the immutable nature as the APIs can be added after startup using apictl.

!!! note

>     If the control plane is enabled, the mounted API projects are not deployed.    

Reviewer: ```suggestion
    If the Control Plane is enabled, the mounted API projects are not deployed.    
```
